### DISCUSSION TITLE
What does lunatic mean?

### DISCUSSION THREAD

**Discussion by tyt2y3** [lunatic-solutions/lunatic#22] [Q&A] [answered]
Out of curiosity, why do you name the library as lunatic? This is totally one of the FAQs!

**Comment by bkolobara** [comment, marked as answer]
I was originally working on a programming language that compiled to Lua, so I named it [Lunatic](https://en.wikipedia.org/wiki/Lunatic). Later I decided to target my own runtime that I called LunaticVm.

In the end I ended up with an actor based language agnostic runtime, but kept the original name. Even it does not have any connections to Lua anymore.

**Comment by tuxiqae** [comment]
Would you consider changing the organization's name from `lunatic-lang` to just `lunatic`?
The former sounds like it's still a language while it seems that your plans have changed.

**Comment by bkolobara** [comment]
I believe the name was taken, otherwise I would have gone for the simpler version.

**Comment by tuxiqae** [comment]
True, what about `Lunatic-VM` or `LunaticVM`?
`Lunatic-Actor` is also available, although I'm not a fan

**Comment by tyt2y3** [comment]
How about LunaticMachine, LunaticRuntime or LunaticWasm?
Just my two cents

**Comment by sleipnir** [comment]
LunaticRT or LunaticVM?

**Comment by bkolobara** [comment]
I have decided to settle on "lunatic-solutions" in the end and also acquired the domain http://lunatic.solutions/.

**Comment by tuxiqae** [comment]
Looking good!
Congratulations!

**Comment by shawntax** [comment]
[Lunatic](https://www.dictionary.com/browse/lunatic) [**loo** -n _uh_ -tik]

_noun_
1. (no longer in technical use; now considered offensive) an insane person.
2. a person whose actions and manner are marked by extreme eccentricity or recklessness.
3. a person legally declared to be of unsound mind and who therefore is not held capable or responsible before the law: a former legal term.
-----

You should be aware that anyone who struggles or has struggled with mental illness, along with disability advocates, ~~will~~ may find the current name of this project offensive and ableist.

**Comment by bkolobara** [comment]
Hi @shawntax, thanks for sharing your concern. I didn't intend to offend anyone with the name, I simply picked it because it originated from the latin word for moon and sounded catchy.

We are an inclusive community and welcome everyone; no matter their background or disabilities. I hope the name doesn't stop anyone from joining or using the project.

**Comment by shawntax** [comment]
Hiya @bkolobara, thanks for responding. I fully believe you have no bad intentions. But regardless of your intent, the fact remains that "lunatic" is an ~~offensive~~ ableist term, potentially offensive to some of those who struggle with and/or care deeply about mental health and the stigma of mental health issues in society. 

I'm coming from an American, English-speaking perspective. The offensiveness of the word is mostly baked in to its older, out-of-style usage. But even its origin from the Latin _luna_ meant ["affected with periodic insanity dependent on the changes of the moon"](https://www.etymonline.com/word/lunatic). So there's really no way to separate the connotation of insanity. 

I know it can be difficult and frustrating to hear this, especially after already investing in a domain and all that. And I'm not a member of your community, so basically as an internet stranger my voice doesn't carry much weight. I'm here only because this repo came up on my GitHub activity feed and it just raised my eyebrow so much that I had to say something. 

**EDIT:** I've updated an instance of the word "offensive" to "ableist", and added a small qualification that not _all_ mentally disabled people will find the term offensive (as TarrenHassman pointed out). But as Lydia X.Z Brown writes in the [Glossary of Ableist Phrases](https://www.autistichoya.com/p/ableist-words-and-terms-to-avoid.html):

> ...many of the words and phrases on this page are not considered slurs, and in fact, may not actually be hurtful, upsetting, retraumatizing, or offensive to many disabled people. They are simply considered ableist (the way that referring to a woman as emotionally fragile is sexist, but not a slur). You're not automatically a bad or evil person/activist if you have used random language on here, but if you have the cognitive/language privilege to adjust your language, it's definitely worthwhile to consider becoming more aware/conscious of how everyday language helps perpetuate ableist ideas and values.

If you stand by your code of conduct and truly care about building an inclusive community, then you would consider [linguistic ableism](https://www.autistichoya.com/2014/02/violence-linguistic-ableism.html) as a valid concern here.

**Comment by TarrenHassman** [comment]
Hi. Mentally disabled person who was recently in rehab. Please stop. The
project name is fine
